code/modules/reagents/recipes/recipes_food_drinks.dm

```diff
@@ -156,6 +156,39 @@
 	for(var/i = 1, i <= created_volume, i++)
 		new /obj/item/reagent_containers/food/snacks/sliceable/butterstick(location)
 
+/datum/chemical_reaction/vanillaicecream
```

Comment:
none of these conflict with milkshake?

Reply:
God damn it. My next best bet was to add temperature requirements but it'd be impossible for a cook to have either instant ice or the means to procure it. How do you suggest we solve this recipe?

Reply:
Alright, I've created a sub-reagent to work for icecream recipes, making Milkshake also be crafted by combining it with milk.
I'm also leaving this reagent for the future if someone wants to add more icecreams, they just have to combine it with their choice of flavor to make icecreams, this should solve all recipe conflicts